demos/notebooks/dqx_quick_start_demo_library.ipynb

```diff
@@ -0,0 +1,209 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install DQX"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install databricks-labs-dqx\n",
+    "%restart_python"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import Required Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import yaml\n",
+    "from databricks.labs.dqx.engine import DQEngine\n",
+    "from databricks.sdk import WorkspaceClient\n",
+    "from pyspark.sql import SparkSession, Row"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configure Test Data\n",
+    "\n",
+    "The result of this next step is `new_users_df`, which represents a dataframe of new users which requires quality validation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark = SparkSession.builder.appName(\"DQX_demo_library\").getOrCreate()\n",
+    "\n",
+    "# Create a sample DataFrame\n",
+    "new_users_sample_data = [\n",
+    "    Row(id=1, age=23, country='Germany'),\n",
+    "    Row(id=2, age=30, country='France'),\n",
+    "    Row(id=3, age=16, country='Germany'), # Invalid -> age - LT 18\n",
+    "    Row(id=None,  age=29, country='France'), # Invalid -> id - NULL\n",
+    "    Row(id=4,  age=29, country=''), # Invalid -> country - Empty\n",
+    "    Row(id=5,  age=23, country='Italy'), # Invalid -> country - not in\n",
+    "    Row(id=6,  age=123, country='France') # Invalid -> age - GT 120\n",
+    "]\n",
+    "\n",
+    "new_users_df = spark.createDataFrame(new_users_sample_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Demonstrating Functions\n",
+    "- `is_not_null_and_not_empty`\n",
+    "- `is_in_range`\n",
+    "- `is_in_list`\n",
+    "\n",
+    "You can find a list of all available built-in checks in the documentation [here](https://databrickslabs.github.io/dqx/docs/reference/quality_rules/) and in the code [here](https://github.com/databrickslabs/dqx/blob/main/src/databricks/labs/dqx/check_funcs.py).\n",
+    "\n",
+    "We are demonstrating creating and validating a set of `Quality Checks` defined declaratively using YAML.\n",
+    "\n",
+    "We can use `validate_checks` to verify that the checks are defined correctly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "checks_from_yaml = yaml.safe_load(\"\"\"\n",
+    "- check:\n",
+    "    function: is_not_null_and_not_empty\n",
+    "    for_each_column:\n",
+    "      - id\n",
+    "      - age\n",
+    "      - country\n",
+    "    criticality: error\n",
```

Comment:
The criticality field must be at the check level

Reply:
Ok, i see this correctly populates the columns `_errors`/`_warnings` but it introduced a bug in `apply_checks_by_metadata_and_split`. 

I will push the code with the error (with the correct configuration) so that you can reproduce on your side.

In the image, review the DF expecting only 'valid' records. After fixing the configs, it includes all records (including failed records). With the **incorrect** configurations it would result on the desired result - only the expected 'valid' records.

![image](https://github.com/user-attachments/assets/c68771fb-9576-4d7c-980c-997fa537f68f)